src/main/java/randoop/main/GenTests.java

```diff
@@ -1335,4 +1358,150 @@ private Path getResourceDirectoryPath(String resourceDirectory) {
   public void incrementSequenceCompileFailureCount() {
     this.sequenceCompileFailureCount++;
   }
+
+  /**
+   * Returns names of classes that the given classes depend on. A class is considered a dependency
+   * if it is a parameter to a method/constructor. Does not return omitted or non-accessible
+   * classes. Does not return dependencies for non-accessible methods/constructors.
+   *
+   * @param classnames names of dependent classes
+   * @param accessibility accessibility predicate
+   * @return classnames of dependencies
+   */
+  public static Set<@ClassGetName String> getDependentClassnamesFromClassnames(
+      Set<@ClassGetName String> classnames, AccessibilityPredicate accessibility) {
+    Set<@ClassGetName String> dependenciesClassnames = new TreeSet<>();
```

Comment:
This code seems to be copied from `addOperationsUsingSignatures`.
Rather than duplicating code, please refactor it to avoid the duplication.  I have done that in https://github.com/randoop/randoop/pull/938.

Reply:
I agree, but this method is private method of `OperationModel` variable, so I couldn't use it. Also, using `OperationModel` for this purposes is not the best approach. So I decided to refactor `getOperationsFromFile` more and moved its logic to new class called `TypedClassOperationProvider`, so now both `GenTests` and `OperationModel` use one implementation located in a new class instead of `GenTests` using `OperationModel`.

Reply:
Can you explain why "using `OperationModel` for this purposes is not the best approach"?  I don't think I understand the reasoning behind that conclusion.

Reply:
From my point of view `OperationModel` is not responsible for reading files and returning objects from them. It was ok when this functionality was used only inside `OperationModel`, but not when we need to perform the same logic in different places of the program. Also, this logic is needed in `GenTests` before `OperationModel` object is created. Maybe better than creating new class I should move this logic to `public static` methods, similar to `OperationModel.readOperations()`?